File: server/Data/Entities/NotificationJob.cs
using System.ComponentModel.DataAnnotations;

namespace Poseidon.Server.Data.Entities;

public class NotificationJob
{
    [Key]
    public Guid NotificationJobId { get; set; } = Guid.NewGuid();

    public Guid EventId { get; set; }
    public Guid RecipientUserId { get; set; }
    public int JobStatusId { get; set; }
    public string Payload { get; set; } = "{}";

    [Required]
    [MaxLength(150)]
    public string Title { get; set; } = string.Empty;

    [Required]
    public string Message { get; set; } = string.Empty;

    [Required]
    [MaxLength(30)]
    public string Channel { get; set; } = "Email";

    public int Attempts { get; set; }
    public DateTimeOffset AvailableAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset? PublisherLockedUntil { get; set; }
    public DateTimeOffset? PublishedAt { get; set; }
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset? ProcessedAt { get; set; }
    public string? LastError { get; set; }

    public Event? Event { get; set; }
    public User? RecipientUser { get; set; }
}

File: mobile/lib/models/notification_job.dart
class NotificationJob {
  const NotificationJob({
    required this.notificationJobId,
    required this.eventId,
    required this.recipientUserId,
    required this.channel,
    required this.title,
    required this.message,
    required this.payloadJson,
    required this.attempts,
    required this.availableAt,
    required this.createdAt,
  });

  final String notificationJobId;
  final String eventId;
  final String recipientUserId;
  final String channel;
  final String title;
  final String message;
  final String payloadJson;
  final int attempts;
  final DateTime availableAt;
  final DateTime createdAt;

  factory NotificationJob.fromJson(Map<String, dynamic> json) =>
      NotificationJob(
        notificationJobId: json['notificationJobId'] as String,
        eventId: json['eventId'] as String,
        recipientUserId: json['recipientUserId'] as String,
        channel: json['channel'] as String,
        title: json['title'] as String,
        message: json['message'] as String,
        payloadJson: json['payloadJson'] as String,
        attempts: json['attempts'] as int,
        availableAt: DateTime.parse(json['availableAt'] as String).toLocal(),
        createdAt: DateTime.parse(json['createdAt'] as String).toLocal(),
      );
}
